tf_files/aws/modules/s3-bucket/cloud.tf

```diff
@@ -151,3 +151,94 @@ resource "aws_iam_instance_profile" "mybucket_writer" {
   name = "bucket_writer_${local.clean_bucket_name}"
   role = "${aws_iam_role.mybucket_writer.id}"
 }
+
+
+
+############################## Added by Fauzi fauzi@uchicago.edu
+
+# let's send the logs to cloudwatch as well
+
+data "aws_cloudwatch_log_group" "logs_destination" {
+  name = "${var.environment}"
+}
+
+# create the policy and role for the role that will be attached to 
+# the trail so it can write cloudwatch
+
+resource "aws_iam_role" "cloudtrail_writer" {
+  name = "cwl_writer_${local.clean_bucket_name}"
+  path = "/"
+
+  assume_role_policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Action": "sts:AssumeRole",
+            "Principal": {
+               "Service": "cloudtrail.amazonaws.com"
+            },
+            "Effect": "Allow",
+            "Sid": ""
+        }
+    ]
+}
+EOF
+}
+
+data "aws_iam_policy_document" "trail_policy" {
+  statement {
+    effect    = "Allow"
+
+    actions = [
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+    ]
+
+    resources = ["${data.aws_cloudwatch_log_group.logs_destination.arn}"]
+  }
+
+}
+
+resource "aws_iam_policy" "trail_writer" {
+  name        = "trail_write_to_cw_${data.aws_cloudwatch_log_group.logs_destination.name}"
+  description = "Read or write     ${data.aws_cloudwatch_log_group.logs_destination.name}"
+  policy      = "${data.aws_iam_policy_document.trail_policy.json}"
+}
+
+resource "aws_iam_role_policy_attachment" "trail_writer_role" {
+  role       = "${aws_iam_role.cloudtrail_writer.name}"
+  policy_arn = "${aws_iam_policy.trail_writer.arn}"
+}
+
+
+# first we need to create a trail in cloudtrail
+
+resource "aws_cloudtrail" "logger_trail" {
+  name                          = "${var.environment}-trail"
+  s3_bucket_name                = "${module.cdis_s3_logs.log_bucket_name}"
+  s3_key_prefix                 = "trailLogs"
```

Comment:
So the parent s3-bucket and cloud-trail both write to the logs bucket?

Reply:
Yes, 

Regular logging, like the module is doing already, would send logs in plain text to the other S3 bucket. Whereas the trail would send similar information but zipped and there is no way to tell CloudTrail to not deliver zipped files.

From my perspective we have two additional options for this redundancy. We could either set a short lifecycle for either one of those (preferably the plain text ones), or we could just not enable bucket logs other than what goes by CloudTrail.

>CloudTrail tracks API access for infrastructure-changing events, in S3 this means creating, seleting, and modifying bucket (S3 CloudTrail docs). It is very focused on API methods that modify buckets.
S3 Server Access Logging provides web server-style logging of access to the objects in an S3 bucket. This logging is granular to the object, includes read-only operations, and includes non-API access like static web site browsing.